src/ApiService/ApiService/onefuzzlib/VmssOperations.cs

```diff
@@ -180,58 +190,74 @@ public class VmssOperations : IVmssOperations {
         return results;
     }
 
+    private record InstanceIdKey(Guid Scaleset, Guid VmId);
+    private Task<string> GetInstanceIdForVmId(Guid scaleset, Guid vmId)
+        => _cache.GetOrCreateAsync(new InstanceIdKey(scaleset, vmId), async _ => {
```

Comment:
should there be expiration on the entries ?

Reply:
It can be configured but I don't think we set it anywhere. It could be nice to have different eviction policies for our different caches.

Also, we should be keeping an eye on memory usage as we're adding caches.

Reply:
we can reuse the TTL value from the [python code](
https://github.com/chkeita/onefuzz/blob/4de2a0cda65de900ee1840e0c6eb219732b52c18/src/api-service/__app__/onefuzzlib/azure/vmss.py#L148) 1min

Reply:
@Porges as discussed in the meeting today i will be taking over this PR and i will add the TTL to unblock prod with hotfix